cmake/modules/boards.cmake

```diff
@@ -51,6 +51,12 @@ include(extensions)
 # If user tries to change the BOARD, the BOARD value is reset to the BOARD_CACHED value.
 zephyr_check_cache(BOARD REQUIRED)
 
+if(SYSBUILD)
+  # Also check if BOARD_ROOT is set in the cache, this is the case when specifying
+  # a custom board directory on the command line when using sysbuild.
+  zephyr_get(BOARD_ROOT SYSBUILD GLOBAL)
```

Comment:
I have a concern that this could collide with the application source relative path handling here:
https://github.com/zephyrproject-rtos/zephyr/blob/a1bb9c9d1736387f9db99c19ce926369bdafda26/cmake/modules/root.cmake#L27-L28

could you check up on that ?

Note, relative paths are only allowed by the user with a `-D`, not when setting the var within another `CMakeLists.txt` file.

Reply:
Fixed